homeassistant/components/yardian/switch.py

```diff
@@ -0,0 +1,78 @@
+"""Support for Yardian integration."""
+from __future__ import annotations
+
+from typing import Any
+
+from homeassistant.components.switch import SwitchEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import YardianUpdateCoordinator
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up entry for a Yardian irrigation switches."""
+    coordinator = hass.data[DOMAIN][config_entry.entry_id]
+    async_add_entities(
+        YardianSwitch(
+            coordinator,
+            config_entry.as_dict(),
+            i,
+        )
+        for i in range(len(coordinator.data.zones))
+    )
+
+
+class YardianSwitch(CoordinatorEntity[YardianUpdateCoordinator], SwitchEntity):
+    """Representation of a Yardian switch."""
+
+    def __init__(self, coordinator: YardianUpdateCoordinator, config, zone_id) -> None:
+        """Initialize a Yardian Switch Device."""
+        super().__init__(coordinator)
+        self.coordinator = coordinator
+        self._zone_id = zone_id
+        self._attr_icon = "mdi:water"
+        self._attr_has_entity_name = True
+        self._attr_unique_id = f"{config['data']['yid']}-{zone_id}"
+        self._attr_device_info = coordinator.device_info
+
+    @property
+    def name(self) -> str:
+        """Return state attributes."""
+        return self.coordinator.data.zones[self._zone_id][0]
```

Comment:
```suggestion
             self._attr_name =  self.coordinator.data.zones[self._zone_id][0]
```

Reply:
But this makes switch's name not responsive when data in coordinator changed. Not the behavior I was expected.

Reply:
Thats fine to leave it as-is since the name can change at run time.